src/components/input-switch/array-field.tsx

```diff
@@ -1,108 +1,326 @@
+import '@isrd-isi-edu/chaise/src/assets/scss/_array-field.scss';
+
 // components
-import ClearInputBtn from '@isrd-isi-edu/chaise/src/components/clear-input-btn';
-import InputField, { InputFieldProps } from '@isrd-isi-edu/chaise/src/components/input-switch/input-field';
-//hooks
-import { useRef } from 'react';
+import { InputFieldProps } from '@isrd-isi-edu/chaise/src/components/input-switch/input-field';
+
 // utils
-import { getSimpleColumnType } from '@isrd-isi-edu/chaise/src/utils/input-utils';
+import InputSwitch from '@isrd-isi-edu/chaise/src/components/input-switch/input-switch';
 import { dataFormats } from '@isrd-isi-edu/chaise/src/utils/constants';
-import { arrayFieldPlaceholder } from '@isrd-isi-edu/chaise/src/utils/input-utils';
-import { windowRef } from '@isrd-isi-edu/chaise/src/utils/window-ref';
-import { hasVerticalScrollbar } from '@isrd-isi-edu/chaise/src/utils/input-utils';
+import { formatDatetime, formatFloat, formatInt, getInputType } from '@isrd-isi-edu/chaise/src/utils/input-utils';
+import React, { useEffect, useRef, useState } from 'react';
+import {
+  DragDropContext, Draggable, DraggableProvided, Droppable, DroppableProps, DroppableProvided, DropResult
+} from 'react-beautiful-dnd';
+import { EventType, useFormContext, useFormState, useWatch } from 'react-hook-form';
+
+// since we're using strict mode, react-beautiful-dnd misbehaves due to multiple renders caused by strict mode
+// this is to guard against it
+const StrictModeDroppable = React.memo(function StrictModeDroppable({ children, ...props }: DroppableProps) {
+  const [enabled, setEnabled] = useState(false);
+  useEffect(() => {
+    const animation = requestAnimationFrame(() => setEnabled(true));
+    return () => {
+      cancelAnimationFrame(animation);
+      setEnabled(false);
+    };
+  }, []);
+  if (!enabled) {
+    return null;
+  }
+  return <Droppable {...props}>{children}</Droppable>;
+});
+
 type ArrayFieldProps = InputFieldProps & {
   /* the type of each element in the array */
   baseArrayType: string,
 };
 
-const ArrayField = (props : ArrayFieldProps): JSX.Element => {
-  const textAreaRef = useRef<HTMLTextAreaElement>(null);
-  const arrayFieldValidation = (value: string) => {
-    if (!value) return;
+type RowItem = {
+  id: number,
+  value: string
+}
+
+type Options = {
+  name?: string,
+  type?: EventType,
+  value?: unknown
+  values?: {
+    [x: string]: any
+  }
+}
+
+const ArrayField = (props: ArrayFieldProps): JSX.Element => {
+  const [itemList, setItemList] = useState<RowItem[]>([])
```

Comment:
Let's make these lines (53 to 60) easier to read,

- Move the deconstructing assignment of `props` to be first line.
- Move the `useFormContext` to be the second line.
- For each state variable definition, add a jsdoc style of comment explaining what each one is
```
/**
 * Explain what this is
 */
const [itemList, setItemList] = useState<RowItem[]>([])
```